File: vibemynight/backend/spring_boot/src/main/java/com/vibemynight/backend/entity/Artist.java
package com.vibemynight.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "artists")
public class Artist extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(name = "photo_url")
    private String photoUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ArtistType type;

    @Column(name = "short_bio", length = 500)
    private String shortBio;

    @Column(name = "full_bio", length = 4000)
    private String fullBio;

    @Column(name = "instagram_url")
    private String instagramUrl;

    @Column(name = "facebook_url")
    private String facebookUrl;

    @Column(name = "youtube_url")
    private String youtubeUrl;

    @Builder.Default
    private boolean featured = false;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ActiveStatus status = ActiveStatus.ACTIVE;
}

File: vibemynight/frontend/flutter_app/lib/models/artist.dart
class Artist {
  final int id;
  final String name;
  final String slug;
  final String? photoUrl;
  final String type;
  final String? shortBio;
  final String? fullBio;
  final String? instagramUrl;
  final String? facebookUrl;
  final String? youtubeUrl;
  final bool featured;
  final String status;

  const Artist({
    required this.id,
    required this.name,
    required this.slug,
    this.photoUrl,
    required this.type,
    this.shortBio,
    this.fullBio,
    this.instagramUrl,
    this.facebookUrl,
    this.youtubeUrl,
    required this.featured,
    required this.status,
  });

  factory Artist.fromJson(Map<String, dynamic> json) => Artist(
        id: (json['id'] as num?)?.toInt() ?? 0,
        name: json['name']?.toString() ?? '',
        slug: json['slug']?.toString() ?? '',
        photoUrl: json['photoUrl']?.toString(),
        type: json['type']?.toString() ?? 'ARTIST',
        shortBio: json['shortBio']?.toString(),
        fullBio: json['fullBio']?.toString(),
        instagramUrl: json['instagramUrl']?.toString(),
        facebookUrl: json['facebookUrl']?.toString(),
        youtubeUrl: json['youtubeUrl']?.toString(),
        featured: json['featured'] == true,
        status: json['status']?.toString() ?? 'ACTIVE',
      );
}
